lib/cset.ml

```diff
@@ -160,3 +164,65 @@ let pick = function
   | [] -> invalid_arg "Re_cset.pick"
   | (x, _) :: _ -> x
 ;;
+
+let cseq c c' = seq (of_char c) (of_char c')
+let rg = cseq
+let char = csingle
+let upper = union_all [ cseq 'A' 'Z'; cseq '\192' '\214'; cseq '\216' '\222' ]
+let clower = offset 32 upper
+let cdigit = cseq '0' '9'
+let ascii = cseq '\000' '\127'
+let cadd c s = add (of_char c) s
+let space = add (of_char ' ') (cseq '\009' '\013')
+let xdigit = union_all [ cdigit; cseq 'a' 'f'; cseq 'A' 'F' ]
+
+let calpha =
+  List.fold_right
+    ~f:cadd
+    [ '\170'; '\181'; '\186'; '\223'; '\255' ]
+    ~init:(union clower upper)
+;;
+
+let calnum = union calpha cdigit
+
+let case_insens s =
+  union_all [ s; offset 32 (inter s upper); offset (-32) (inter s clower) ]
+;;
+
+let cword = cadd '_' calnum
+let notnl = diff cany (csingle '\n')
+let nl = csingle '\n'
+
+let set str =
+  let s = ref empty in
+  for i = 0 to String.length str - 1 do
+    s := union (csingle str.[i]) !s
+  done;
+  !s
+;;
+
+let blank = set "\t "
+
+(* CR rgrinberg: this [lower] doesn't match [clower] *)
```

Comment:
cc @vouillon. I assume this distinction is unintentional. Note that `upper` and `cupper` were the same and I merged them in this PR.